src/main/java/org/apache/pirk/query/wideskies/Query.java

```diff
@@ -136,15 +136,17 @@ public void generateExpTable()
       public void accept(BigInteger element)
       {
         Map<Integer,BigInteger> powMap = new HashMap<>(maxValue); // <power, element^power mod N^2>
-        for (int i = 0; i <= maxValue; ++i)
+        BigInteger value = BigInteger.ONE;
+        powMap.put(0, value);
+        for (int i = 1; i <= maxValue; ++i)
         {
-          BigInteger value = ModPowAbstraction.modPow(element, BigInteger.valueOf(i), NSquared);
+          value = value.multiply(element).mod(NSquared);
```

Comment:
... to avoid the need to compute the power of each big integer.  The point of the PR is to just use multiplication.

For example, the code may be creating an expTable with 256 powers of a number of big integers.

In the current implementation using ModPowAbstration that is achieved by computing all 256 powers individually, i.e. element^0, element^1, ... element^255.  Clearly as we build the table we know the value for (element-1)^n so we can simply remember the previous value and multiply it by element to find the next value.

This runs faster in my benchmarking.